src/evict/evict_page.c

```diff
@@ -675,63 +677,71 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
             LF_SET(WT_REC_SCRUB);
     }
 
-    /* Acquire a snapshot if coming through eviction thread route. */
+    /*
+     * Acquire a snapshot if coming through eviction thread route. Also, if we have entered eviction
+     * through application threads and we have a transaction snapshot, we will use our existing
+     * snapshot to evict pages that are not globally visible based on last_running transaction.
+     * Avoid using snapshots when application transactions are in final stages of commit or rollback
+     * as they have already released the snapshot. Otherwise, it becomes harder in later part of the
+     * code to detect updates the belonged to the last running application transaction.
```

Comment:
...Otherwise it becomes harder in the later part...
...to detect updates that belonged...

Reply:
Fixed in 65be4b5.